adapters/http4s/src/main/scala/caliban/Http4sAdapter.scala

```diff
@@ -292,7 +292,7 @@ object Http4sAdapter {
    */
   def provideSomeLayerFromRequest[R <: Has[_], R1 <: Has[_]](
     route: HttpRoutes[RIO[R with R1, *]],
-    f: Request[RIO[R, *]] => TaskLayer[R1]
+    f: Request[RIO[R, *]] => RLayer[R, R1]
```

Comment:
How about `provideLayerFromRequest` above? We can change it the same way, right?

Reply:
I tried it and now I looked at method type definition and I'm not sure if it is possible. Look
```
  def provideLayerFromRequest[R <: Has[_]](route: HttpRoutes[RIO[R, *]], f: Request[Task] => TaskLayer[R])(
    implicit tagged: Tag[R]
  ): HttpRoutes[Task] =
```
```
type HttpRoutes[F[_]] = Http[OptionT[F, *], F]
```
```
type Http[F[_], G[_]] = Kleisli[F, Request[G], Response[G]]
```
This means that when I use HttpRoutes, the inner (request body) and outer (request returning) effect needs to be same. So function `f: Request[F]` needs to have same effect as result value which for sure must be `HttpRoutes[Task]`. So I can't change effect for Request body. I would work if I change result type to something like Http[OptionT[RIO[R, *], *],  Task] so outer and inner effect would not be the same. But it might be more complicated for users. But I might be wrong, so pls correct me if so.

Reply:
Oh I see. Let's merge this one then.